src/winter_onboarding_2021/fleet_management_service/handlers/fleet.clj

```diff
@@ -14,11 +15,33 @@
                     :created-by user-id}]
     (if (s/valid? ::specs/fleet-form fleet-data)
       (let [fleet-id (:fleets/id (fleet-model/create fleet-data))]
-        (->  (flash-msg "Fleet created successfully!" true)
+        (->  (utils/flash-msg "Fleet created successfully!" true)
              (merge (response/redirect (format "/fleets/%s" (str fleet-id))))))
-      (->  (flash-msg "Could not create fleet, try again!" false)
+      (->  (utils/flash-msg "Could not create fleet, try again!" false)
            (merge (response/redirect "/fleets/new"))))))
 
 (defn new [_]
   {:title "Create fleet"
    :content (views/create-fleet)})
+
+(defn assoc-managers [fleet]
+  (assoc fleet
+         :managers
+         (fleet-model/managers fleet)))
+
+(defn show-fleets [request]
+  (let [user-id (get-in request [:user :users/id])
+        {:keys [page]} (:params request)
+        page-size (config/get-page-size)
+        current-page (Integer/parseInt (or page "1"))
+        offset (utils/calc-offset page-size current-page)
+        fleets (fleet-model/select-fleets-for-admin user-id
+                                                    offset
+                                                    page-size)
+        fleets-count (:count (first (fleet-model/total)))
+        show-next-page? (<= (* current-page page-size) fleets-count)
+        fleets-with-managers (map assoc-managers fleets)]
```

Comment:
Note: Map is lazy. When you map assoc-managers on fleets, the query to fetch managers will be lazily executed.